amespahdbpythonsuite/mcfitted.py

```diff
@@ -0,0 +1,256 @@
+#!/usr/bin/env python3
+
+from __future__ import annotations
+from typing import Optional
+
+import numpy as np
+import matplotlib.pyplot as plt
+
+from amespahdbpythonsuite.amespahdb import AmesPAHdb
+from amespahdbpythonsuite.spectrum import Spectrum
+
+message = AmesPAHdb.message
+
+
+class Mcfitted(Spectrum):
+    """
+    AmesPAHdbPythonSuite mcfitted class.
+    Contains methods to fit and plot the input spectrum using a Monte Carlo approach.
+
+    """
+
+    def __init__(self, d: Optional[dict] = None, **keywords) -> None:
+        super().__init__(d, **keywords)
+        self.__set(d, **keywords)
+
+    def set(self, d: Optional[dict] = None, **keywords) -> None:
+        """
+        Calls class: :class:`amespahdbpythonsuite.transitions.Transitions.set` to parse keywords.
+
+        """
+        Spectrum.set(self, d, **keywords)
+        self.__set(d, **keywords)
+
+    def __set(self, d: Optional[dict] = None, **keywords) -> None:
+        """
+        Populate data dictionary helper.
+
+        """
+        if isinstance(d, dict):
+            if d.get("type", "") == self.__class__.__name__:
+                if "observation" not in keywords:
+                    self.observation = d["observation"]
+                if "mcfits" not in keywords:
+                    self.mcfits = d["mcfits"]
+                if "mcweights" not in keywords:
+                    self.mcweights = d["mcweights"]
+                if "mcclasses" not in keywords:
+                    self.mcclasses = d["mcclasses"]
+                if "mcresults" not in keywords:
+                    self.mcresults = d["mcresults"]
+                if "mcpredicted" not in keywords:
+                    self.mcpredicted = d["mcpredicted"]
+                if "method" not in keywords:
+                    self.method = d["method"]
+
+        self.observation = keywords.get("observation", None)
+        self.mcfits = keywords.get("mcfits", list())
+        self.mcweights = keywords.get("mcweights", list())
+        self.mcclasses = keywords.get("mcclasses", list())
+        self.mcresults = keywords.get("mcresults", list())
+        self.mcpredicted = keywords.get("mcpredicted", False)
+        self.method = keywords.get("method", "")
+
+    def get(self) -> dict:
+        """
+        Calls class: :class:`amespahdbpythonsuite.transitions.Transitions.get`.
+        Assigns class variables from inherited dictionary.
+
+        """
+        d = Spectrum.get(self)
+        d["type"] = self.__class__.__name__
+        d["observation"] = self.observation
+        d["mcfits"] = self.mcfits
+        d["mcweights"] = self.mcweights
+        d["mcclasses"] = self.mcclasses
+        d["mcresults"] = self.mcresults
+        d["mcpredicted"] = self.mcpredicted
+        d["method"] = self.method
+
+        return d
+
+    def stats(self, results):
```

Comment:
For consistency, the same interface as offered for Fitted should be offered and employ 'Lazy' instantiation for these methods. At some point, that should also be done for Fitted itself-the IDL-implementation already does this.